src/error.rs

```diff
@@ -19,126 +19,109 @@ pub type Result<T> = std::result::Result<T, Error>;
-    InvalidParameter(&'static str),
-    /// No node available in the synced node pool
-    #[error("No synced node available")]
-    SyncedNodePoolEmpty,
-    /// Error on reaching quorum
-    #[error("Failed to reach quorum: {quorum_size} < {minimum_threshold}")]
-    QuorumThresholdError {
-        /// The current quorum size.
-        quorum_size: usize,
-        /// The minimum quorum threshold.
-        minimum_threshold: usize,
-    },
+    /// Rw lock failed.
+    #[error("rw lock failed")]
+    PoisonError,
+    /// PoW error
+    #[error("{0}")]
+    Pow(String),
+    /// Prefix hex string convert error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    PrefixHexError(#[from] prefix_hex::Error),
     /// Error on quorum because not enough nodes are available
-    #[error("Not enough nodes for quorum: {available_nodes} < {minimum_threshold}")]
+    #[error("not enough nodes for quorum: {available_nodes} < {minimum_threshold}")]
     QuorumPoolSizeError {
         /// The number of nodes available for quorum.
         available_nodes: usize,
         /// The minimum quorum threshold.
         minimum_threshold: usize,
     },
-    /// Error on API request
-    #[error("Node error: {0}")]
-    NodeError(String),
-    /// Prefix hex string convert error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    FromHexError(#[from] prefix_hex::Error),
-    /// Block types error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    BlockError(#[from] bee_block::Error),
-    /// Unpack error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    UnpackError(#[from] packable::error::UnpackError<bee_block::Error, UnexpectedEOF>),
-    /// Block dtos error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    BlockDtoError(#[from] bee_block::DtoError),
-    /// Bee rest api error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    BeeRestApiError(#[from] bee_api_types::error::Error),
-    /// The block doensn't need to be promoted or reattached
-    #[error("Block ID `{0}` doesn't need to be promoted or reattached")]
-    NoNeedPromoteOrReattach(String),
-    /// The block cannot be included into the Tangle
-    #[error("Block ID `{0}` couldn't get included into the Tangle")]
-    TangleInclusionError(String),
-    /// Mqtt client error
-    #[cfg(feature = "mqtt")]
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    MqttClientError(#[from] rumqttc::ClientError),
-    /// Invalid MQTT topic.
-    #[error("The MQTT topic {0} is invalid")]
-    InvalidMqttTopic(String),
-    /// MQTT connection not found (all nodes MQTT's are disabled)
-    #[error("MQTT connection not found (all nodes have the MQTT plugin disabled)")]
-    MqttConnectionNotFound,
-    /// IO error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    IoError(#[from] std::io::Error),
```

Comment:
What was the motive to remove this one?

Reply:
It wasn't used in the library, only in the examples, no?

Reply:
Hmm yeah it wasn't used by *ourselves* but we're still a library, it just makes it easier for consumers to use `?` I guess

Reply:
Yes, but we shouldn't add support for all kind of possible errors, just so it's easier to use from us, even so we don't need this at all?

Reply:
Yeah I understand the point of view. I would have said that IO at least is a good one to handle but let's forget about it 👍🏻